# Patch jupyter_core to help jupyter find the correct installation path.
#
# If jupyter is installed by EasyBuild, jupyter is not in the same place as python.
# `EB_ENV_JUPYTER_ROOT` is used to indicate where jupyter are and should be set to
# the installation path in easyconfigs.
# Avoid using `ENV_JUPYTER_PATH` and `ENV_CONFIG_PATH`. Otherwise user configuration
# has lower priority and cannot save their own settings.
# Author: Chia-Jung Hsu, 2024-01-31
# Updated for jupyter-core 5.9.1 by Pavel Tomanek (Ugent/Inuits)

--- a/jupyter_core/paths.py
+++ b/jupyter_core/paths.py
@@ -196,7 +196,11 @@
         ]
 
-ENV_JUPYTER_PATH: list[str] = [str(Path(sys.prefix, "share", "jupyter"))]
+if os.environ.get("EB_ENV_JUPYTER_ROOT"):
+    EB_ENV_JUPYTER_ROOT = [p.rstrip(os.sep) for p in os.environ["EB_ENV_JUPYTER_ROOT"].split(os.pathsep)]
+    ENV_JUPYTER_PATH = [str(Path(p, "share", "jupyter")) for p in EB_ENV_JUPYTER_ROOT]
+else:
+    ENV_JUPYTER_PATH: list[str] = [str(Path(sys.prefix, "share", "jupyter"))]
 
 def jupyter_path(*subdirs: str) -> list[str]:
     """Return a list of directories to search for data files.
@@ -276,7 +280,11 @@
     return paths
 
-ENV_CONFIG_PATH: list[str] = [str(Path(sys.prefix, "etc", "jupyter"))]
+if os.environ.get("EB_ENV_JUPYTER_ROOT"):
+    EB_ENV_JUPYTER_ROOT = [p.rstrip(os.sep) for p in os.environ["EB_ENV_JUPYTER_ROOT"].split(os.pathsep)]
+    ENV_CONFIG_PATH = [str(Path(p, "etc", "jupyter")) for p in EB_ENV_JUPYTER_ROOT]
+else:
+    ENV_CONFIG_PATH: list[str] = [str(Path(sys.prefix, "etc", "jupyter"))]
 
 if use_platform_dirs():
     if os.name == "nt" and not _use_programdata: